File: Crio/crio_programming_interview_problems-master/StringRotation/StringRotation.py

def is_s1_rotation_of_s2(s1, s2):
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    l1 = len(s1)
    l2 = len(s2)
    
    if l1 != l2:
        return False


    for i in range(l1):
      p1 = s1[0:i]
      p2 = s1[i:]
      if s2 == p2 + p1:
          return True

# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
 
    return False   

def main():
    (s1, s2) = input().strip().split()
    if is_s1_rotation_of_s2(s1, s2):
        print("Yes")
    else:
        print("No")

if __name__ == '__main__':
    main()
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    s1 = 'a' * 700 + 'b' * 300
    s2 = 'b' * 100 + 'a' * 700 + 'b' * 200
    assert(is_s1_rotation_of_s2(s1, s2))
    s2 = 'a' * 400 + 'b' * 300 + 'a' * 300
    assert(is_s1_rotation_of_s2(s1, s2))
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

File: Crio/crio_programming_interview_problems-master/StringRotation/StringRotation.cpp
#include<bits/stdc++.h>
using namespace std;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
string getSubString(int l , int r , string s) {
	string ret = "";
	for(int i = l ; i <= r ; i++) {
		ret += s[i];
	}
	return ret;
}
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

bool isRotation(string s1 , string s2) {
// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	if(s1.size() != s2.size()) {
		return false;
	}
	for(int i = s1.size()-1 ; i >= 0 ; i--) {
		if(getSubString(i,s1.size()-1,s1) == getSubString(0,s1.size()-1-i,s2)) {
			if(getSubString(0,i-1,s1) == getSubString(s1.size()-i,s2.size()-1,s2)) {
				return true;	
			}
			
		}
	}
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
	return false;
}

int main()
{
	string s1 , s2;
	cin >> s1 >> s2; 
	bool answer = isRotation(s1,s2);
	if(answer) {
		cout << "Yes";
	} else {
		cout << "No";
	}
	return 0;
}
